src/Compilers/CSharp/Portable/Compilation/SyntaxTreeSemanticModel.cs

```diff
@@ -701,7 +701,8 @@ private MemberSemanticModel GetMemberModel(int position)
                         var constructorDecl = (ConstructorDeclarationSyntax)memberDecl;
                         outsideMemberDecl =
                             !LookupPosition.IsInConstructorParameterScope(position, constructorDecl) &&
-                            !LookupPosition.IsInParameterList(position, constructorDecl);
+                            !LookupPosition.IsInParameterList(position, constructorDecl) &&
+                            !LookupPosition.IsInExpressionBody(position, constructorDecl.GetExpressionBodySyntax(), constructorDecl.SemicolonToken);
```

Comment:
Shouldn't the switch condition for the `SyntaxKind.MethodDeclaration` etc. be changed to `IsInBody` as well?